pkg/compiler/check.go

```diff
@@ -809,9 +813,13 @@ func (comp *compiler) checkTypeBasic(t *ast.Type, desc *typeDesc, flags checkFla
 		comp.error(t.Pos, "%v can't be type alias target", t.Ident)
 		return
 	}
-	if flags&checkIsResourceBase != 0 &&
+	if (flags&checkIsResourceBase != 0 || flags&checkIsTypedefWithBase != 0) &&
```

Comment:
This seems to be checking wrong arg. If I do:
```
type templ10[BASE] const[0, BASE]
foo$209(a ptr[in, templ10[int8]])
```
I get "const can't be template base (int types can)". But the code looks correct.

Reply:
Yes, this was completely wrong. The new version should be ok; I've added more tests, including your example.